### ISSUE TITLE
Encryption Support?

### ISSUE THREAD

**Issue by majmongoose** [hmatuschek/qdmr#200]
My radio (DMR 6x2) finds dual use on ham bands and commercial radio bands in the US, where we are permitted to use encryption on commercial. Is there any plans to implement AES encryption programming? It is supported in the official CPS.

**Comment by hmatuschek** [comment]
Yes, it is planned but as it is not HAM radio related, it has a somewhat low priority.

**Comment by hmatuschek** [comment]
Ok I use this issue as an excuse to finalize the implementation of the device specific settings. That is, encryption will be implemented as an extension in the `encryption` branch. Things needed:

- [x] Implement EncryptionExtension
- [x] Implement CommercialChannelExtension
- [ ] Document extensions in manual
- [ ] Implement handling of ConfigObjectLists in ExtensionWrapper
- [x] Implement handling of ConfigObjectReference in ExtensionWrapper
- [ ] Implement encoding/decoding of encryption in Radioddity Codeplugs
- [x] Implement encoding/decoding of encryption in TyT Codeplugs
- [ ] Implement encoding/decoding of encryption in AnyTone Codeplugs

**Comment by audriusb** [comment]
i like that it's there, but i can only see 128 bit support. both Anytone 878 and TYT 390 support AES256 now.

**Comment by falkrist** [comment]
Hi! Just wandering on the same, if there any plans to add AES256 support also? It's supported in official CPS for long time, would be nice to have this also in qdmr.

**Comment by hmatuschek** [comment]
Still need to update encryption support for TyT devices...
